### PR TITLE
list, create, edit, delete programming exercises

### PR DESCRIPTION
**Pull request by VSchlattinger**
Ported functionality for listing, creating, editing, deleting exercises to programming exercises and added dropdown to select which kind of exercises to view from Course list in "Entities" View.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/exerciseapp/web/rest/ProgrammingExerciseResource.java [hunk 1]
```diff
@@ -79,19 +113,35 @@ public ProgrammingExerciseResource(ProgrammingExerciseRepository programmingExer
...
+     * GET  /courses/:courseId/exercises : get all the exercises.
+     *
+     * @return the ResponseEntity with status 200 (OK) and the list of programmingExercises in body
+     */
+    @GetMapping(value = "/courses/{courseId}/programming-exercises")
+    @PreAuthorize("hasAnyRole('TA', 'ADMIN')")
+    @Timed
+    @Transactional(readOnly = true)
+    public List<ProgrammingExercise> getProgrammingExercisesForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get a page of Exercises");
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/exerciseapp/web/rest/ProgrammingExerciseResource.java]: The log statement is quite generic --> would be better to be specific with "Programming Exercises"

File: src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java [hunk 2]
```diff
@@ -79,11 +83,26 @@ public QuizExerciseResource(QuizExerciseRepository quizExerciseRepository) {
...
+     * GET  /courses/:courseId/exercises : get all the exercises.
+     *
+     * @return the ResponseEntity with status 200 (OK) and the list of programmingExercises in body
+     */
+    @GetMapping(value = "/courses/{courseId}/quiz-exercises")
+    @PreAuthorize("hasAnyRole('TA', 'ADMIN')")
+    @Timed
+    @Transactional(readOnly = true)
+    public List<QuizExercise> getQuizExercisesForCourse(@PathVariable Long courseId) {
+        log.debug("REST request to get a page of Exercises");
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/exerciseapp/web/rest/QuizExerciseResource.java]: The log statement is quite generic --> would be better to be specific with "Quiz Exercises"

File: src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js [hunk 3]
```diff
@@ -13,7 +13,7 @@
             parent: 'entity',
             url: '/programming-exercise',
             data: {
-                authorities: ['ROLE_USER'],
+                authorities: ['ROLE_ADMIN', 'ROLE_TA'],
                 pageTitle: 'artemisApp.programmingExercise.home.title'
```
Comment (krusche) [inline src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js]: why did the authority change here?

File: src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js [hunk 4]
```diff
@@ -167,6 +257,106 @@
                     $state.go('^');
                 });
             }]
+        })
+        .state('programming-exercise-for-course.delete', {
+            parent: 'programming-exercise-for-course',
+            url: '/{id}/delete',
+            data: {
+                authorities: ['ROLE_ADMIN', 'ROLE_TA']
```
Comment (krusche) [inline src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js]: delete an exercise is something that only admins should be able to do, please remove the ROLE_TA here

File: src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js [hunk 5]
```diff
@@ -13,7 +13,7 @@
             parent: 'entity',
             url: '/quiz-exercise',
             data: {
-                authorities: ['ROLE_USER'],
+                authorities: ['ROLE_ADMIN', 'ROLE_TA'],
```
Comment (krusche) [inline src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js]: same as above.. I am note sure what this js function is supposed to do, but are you sure that normal users should not be able to do it?

File: src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js [hunk 6]
```diff
@@ -144,7 +171,7 @@
             parent: 'quiz-exercise',
             url: '/{id}/delete',
             data: {
-                authorities: ['ROLE_USER']
+                authorities: ['ROLE_ADMIN', 'ROLE_TA']
```
Comment (krusche) [inline src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js]: delete only for admins

File: src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js [hunk 7]
```diff
@@ -163,7 +190,31 @@
                     $state.go('^');
                 });
             }]
-        });
+        })
+            .state('quiz-exercise-for-course.delete', {
+                parent: 'quiz-exercise',
+                url: '/{id}/delete',
+                data: {
+                    authorities: ['ROLE_ADMIN', 'ROLE_TA']
```
Comment (krusche) [inline src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js]: delete only for admins

File: src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js (hunk 3 above)
Comment (VSchlattinger) [reply src/main/webapp/app/entities/programming-exercise/programming-exercise.state.js]: the `ROLE_USER` was automatically generated by jHipster when we created the programming-exercise entity. I looked at `src/main/webapp/app/entities/exercise/exercise.state.js` and there the authorities are `ROLE_ADMIN` and `ROLE_TA`, so I copied that over.

File: src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js (hunk 5 above)
Comment (VSchlattinger) [reply src/main/webapp/app/entities/quiz-exercise/quiz-exercise.state.js]: This state is for listing all quiz-exercises in the "Entities" View. Normal users shouldn't have access to that part anyway.

**Pull request merged**